openmc/tallies.py

```diff
@@ -1410,6 +1432,765 @@ def export_results(self, filename='tally-results', directory='.',
             # Pickle the Tally results to a file
             pickle.dump(tally_results, open(filename, 'wb'))
 
+    def _outer_product(self, other, new_tally, binary_op):
+        """Combines filters, scores and nuclides with another tally.
+
+        This is a helper method for the tally arithmetic methods. The filters,
+        scores and nuclides from both tallies are enumerated into all possible
+        combinations and expressed as CrossFilter, CrossScore and
+        CrossNuclide objects in the new derived tally.
+
+        Parameters
+        ----------
+        other : Tally
+            The tally on the right hand side of the outer product
+        new_tally: Tally
+            The new tally to represent the outer product
+        op : str
+            The binary operation in the outer product ('+', '-', '*', '/', '^')
+
+        """
+
+        new_tally.name = '({0} {1} {2})'.format(self.name, binary_op, other.name)
+
+        if self.estimator == other.estimator:
+            new_tally.estimator = self.estimator
+        if self.with_summary and other.with_summary:
+            new_tally.with_summary = self.with_summary
+        if self.num_realizations == other.num_realizations:
+            new_tally.num_realizations = self.num_realizations
+
+        # Generate filter "outer products"
+        if self.filters == other.filters:
+            for self_filter in self.filters:
+                new_tally.add_filter(self_filter)
+        else:
+            all_filters = [self.filters, other.filters]
+            for self_filter, other_filter in itertools.product(*all_filters):
+                new_filter = CrossFilter(self_filter, other_filter, binary_op)
+                new_tally.add_filter(new_filter)
+
+        # Generate score "outer products"
+        if self.scores == other.scores:
```

Comment:
Oh nevermind, I just thought about this again. The simple equality should be ok.